### ISSUE TITLE
Fixtures not loading from bundle

### ISSUE THREAD

**Issue by timtimosh** [hautelook/AliceBundle#389]
Hello, here is the problem, i`m trying to load fixtures from my bundle
`bin/console hautelook:fixtures:load -b CatalogBundle`
but they don`t loading from CatalogBundle, they load from fixtures_path

```
hautelook_alice:
    fixtures_path: 'Resources/fixtures'
```
"symfony/symfony": "3.4.*",
"hautelook/alice-bundle": "~v2.0"

**Comment by theofidry** [comment]
Indeed that looks like a valid bug. Could you try to re-create a failing test on the repo?

**Comment by timtimosh** [comment]
you want me to open my project repo so you could re-create it?

**Comment by theofidry** [comment]
That or if possible as this would make things even easier: fork this repository and add a test case where this is failing

**Comment by curry684** [comment]
I ran into an issue which you my also be having: I had my fixtures under `Resources/fixtures/orm`, but somewhere recently the scan stopped being recursive, so they weren't detected. Moving them one directory up fixed it.

**Comment by umpirsky** [comment]
Maybe [at this point](https://github.com/hautelook/AliceBundle/pull/262/files#diff-7a1fda4a7dbf099d7959d05c54b894f7L65).

**Comment by SanderVerkuil** [comment]
@umpirsky You are correct, removing the `->depth(0)` part locally in the vendor folder solves the issue.

This issue occurs with the following folder structure:

```
|-app
|-fixtures
|---dev
|-----userdata
|--------user.yaml
|--------account.yaml
|------products
|--------categories.yaml
|--------products.yaml
|-----test.yaml
```

In this example, only `tests.yaml` is found by the finder, because of the `->depth(0)` function.

@theofidry Is there a reason why you limited the depth of the `EnvDirectoryLoader` to 0?

**Comment by theofidry** [comment]
There was a reason but I don't which one. Maybe try it and run the tests and see what happens, it may give the reason. But maybe the reason is not valid anymore and the behaviour could be changed...

**Comment by SanderVerkuil** [comment]
The reason was that when you load the environmentless files and you load them with an unlimited depth, all the files of all the environments are also loaded.

So how would we determine the available environments and exclude them in the matcher?

**Comment by umpirsky** [comment]
Why don't we make depth configurable and optional?

**Comment by SanderVerkuil** [comment]
Because if you indeed use different fixtures per environment, the
'environment-less' fixtures should exlude those in an environment. Dev,
Prod and Test are default fixtures, but others might include vagrant or
homesteead as other environments, and loading the fixtures for environment
'dev', both the environment-less fixtures and the dev fixtures should be
loaded, but not the prod, test and other configured environments.

Are the available environments exposed in the symfony container? Than we
could just exclude those folders when loading the fixtures.

Op vr 18 mei 2018 om 12:58 schreef Saša Stamenković <
notifications@github.com>:

**Comment by umpirsky** [comment]
I guess they are, but it's not only about environments. Sometimes you want to organize folder structure in a different way, and you need depth.

**Comment by theofidry** [comment]
when do we actually have environment less? There is always an environment involved since the application runs in a default environment

**Comment by SanderVerkuil** [comment]
I think that the intention of the environment-less files was to always load them, like configuration in the database that should always be present.

So for every load, both the 'environment-less' files and the files of the specified environment are loaded

**Comment by theofidry** [comment]
yes that's my interpretation as well. So I think an improvement is not going recursively for the environment less fixtures but doing so for the environment specific fixtures

**Comment by theofidry** [comment]
Revisiting this, I wonder if this environment base loading still makes any sense (as well as the bundle one). Maybe it would be a lot easier to just load the `hautelook_alice.fixtures_path` by default and alternatively be able to load an arbitrary directory as @alcalyn mentionned

**Comment by alcalyn** [comment]
I like the folder recursivity you talked about. What I could expect is to have something like:

```
fixtures/
    status.yml
    any-always-needed-fixtures.yml
    demo/
        user.yml
        sample-data-for-demo.yml
```

Then I can run:

``` bash
# Only fixtures in root
bin/console hautelook:fixtures:load

# Or also add folder from folder
bin/console hautelook:fixtures:load demo
```

In my use case, I don't need anything related to application environment.
